### PR TITLE
UX: Provide full diff for code samples

### PR DESCRIPTION
**Pull request by keradus**
Our code samples are super condensed and each line of them has a purpose.

For example, diff for first code sample of `native_function_invocation` is:
```
 * Example #1. Fixing with the default configuration.
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -2,9 +2,9 @@
    
    function baz($options)
    {
   -    if (!array_key_exists("foo", $options)) {
   +    if (!\array_key_exists("foo", $options)) {
            throw new \InvalidArgumentException();
        }
    
   -    return json_encode($options);
   +    return \json_encode($options);
    }
   
   ----------- end diff -----------
```

The second sample is this:
```
 * Example #2. Fixing with configuration: ['exclude' => ['json_encode']].
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -2,7 +2,7 @@
    
    function baz($options)
    {
   -    if (!array_key_exists("foo", $options)) {
   +    if (!\array_key_exists("foo", $options)) {
            throw new \InvalidArgumentException();
        }
    
   
   ----------- end diff -----------
```

in fact, they are using very same input code, just different configuration. But in later, we don't know that, as we don't see `json_encode` in diff at all.

------------------------

Proposed PR would change 2nd example into:
```
 * Example #2. Fixing with configuration: ['exclude' => ['json_encode']].
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -1,10 +1,10 @@
    <?php
    
    function baz($options)
    {
   -    if (!array_key_exists("foo", $options)) {
   +    if (!\array_key_exists("foo", $options)) {
            throw new \InvalidArgumentException();
        }
    
        return json_encode($options);
    }
   
   ----------- end diff -----------
```

### CODE REVIEW

**Comment by SpacePossum** [comment]
:+1: for the idea, we could even create a real `full differ` on `sebastianbergmann/diff` one day.

**Comment by keradus** [comment]
What would be the difference in generated output between differ in this PR and that `full differ` you imagine about? TBH, not sure if there is a need - for udiffer, it was important for us, we wanted to use it heavily and in the and, a lot of ppl were looking for udiffer implementation in PHP.
But `full differ` ? Well, there is not that much usage of it in the wild, and here, at our `code samples` case, it's super edge case.
Long story short - not sure if it's worth to spend time on it
